Rework command wrapper macros

michael-grunder opened the pull request:
This commit reworks the `Redis` and `RedisCluster` command wrapper macros into static functions.

The changes result in around a ~30% reduction in binary size and either no change in performance or a slight improvement depending on workload.

michael-grunder committed:
Fix dead assignment

michael-grunder committed:
Rework `CLUSTER_PROCESS_CMD` to use an underlying function

In theory this should reduce PhpRedis' code size and likely doesn't
affect performance in a measurable way.

michael-grunder committed:
Rework `CLUSTER_PROCESS_KW_CMD` to be a small wrapper macro + function

This commit is similar to the last one reworking processing keyword
commands to work to use a function instead of a big multiline macro.

michael-grunder committed:
Make `CLUSTER_ENQUEUE_RESPONSE` a static function

michael-grunder committed:
Remove dead macro `CLUSTER_BUILD_CMD`

michael-grunder committed:
Rewowrk `CLUSTER_FREE_QUEUE` as a static function

michael-grunder committed:
Rework `CLUSTER_RESET_MULTI` to be a static function

michael-grunder committed:
Rework `REDIS_PROCESS_CMD` into a static function

michael-grunder committed:
Rework `REDIS_PROCESS_KW_CMD` as a function

michael-grunder committed:
Remove dead legacy code.

michael-grunder committed:
Rework `REDIS_PROCESS_REQUEST` to be a function.

michael-grunder committed:
Remove wrapper macro which hides branching logic

This wrapper macro implicitly defines an `} else {` block but this is
not clear at the callsite which obsures what is actually going on.

There's no real advantage to the wrapping macro. Instead just call the
underlying macro in an explicit else branch.

michael-grunder committed:
Rework `REDIS_SAVE_CALLBACK` to be a function.

michael-grunder committed:
Change `int` flags to `zend_bool` since we only use them as booleans

michael-grunder committed:
Remove pointless casts

You never have to explicitly cast between `void*` and any other pointer
type.

michael-grunder committed:
Reduce to < 80 chars

michael-grunder committed:
Use ZEND_STRL instead of literal, len

yatsukhnenko commented on redis_cluster.c:
```diff
@@ -43,6 +43,119 @@ zend_class_entry *redis_cluster_exception_ce;
+    redisClusterNode *node;
+    ZEND_HASH_FOREACH_PTR(c->nodes, node) {
+        if (node == NULL)
+            break;
```
```suggestion
            continue;
```

michael-grunder replied:
Yeah continue is better here. I saw this too when changing it from the macro which currently calls break. My guess is this is just a bug that has never happened because we never add a null node?

yatsukhnenko reviewed (approved):
This is great!

michael-grunder committed:
Use continue not break if we get a NULL node

This is likely to never happen but just skipping NULL nodes is better than aborting the reset.

Co-authored-by: Pavlo Yatsukhnenko <yatsukhnenko@users.noreply.github.com>
